src/ProductConstructionService/ProductConstructionService.Deployment/DeploymentOptions.cs

```diff
@@ -28,4 +36,26 @@ public class DeploymentOptions
     public required bool IsCi { get; init; }
     [Option("entraAppId", Required = true, HelpText = "Entra app ID")]
     public required string EntraAppId { get; init; }
+    [Option("redisConnectionString", Required = true, HelpText = "Redis Cache connection string")]
+    public required string RedisConnectionString { get; init; }
+
+    public async Task<IServiceCollection> RegisterServices(IServiceCollection services)
+    {
+        services.AddLogging(logging => logging.AddConsole());
+
+        services.AddTransient<IProcessManager>(sp => new ProcessManager(sp.GetRequiredService<ILogger<ProcessManager>>(), string.Empty));
+        DefaultAzureCredential credential = new();
+        services.AddTransient(_ => credential);
```

Comment:
This is fishy, why not just register a singleton right away?

Reply:
we use the credential later to configure the `ConfigurationOptions`. The other option would be to do something like 
```
services.AddSingleton(sp =>
    ConfigurationOptions.Parse(RedisConnectionString)
        .ConfigureForAzureWithTokenCredentialAsync(
            sp.GetRequiredService<DefaultAzureCredential>()).GetAwaiter().GetResult());
```